### ISSUE TITLE
GPU actors slower than Serial execution

### ISSUE THREAD

**Issue by jakezhaojb** [ray-project/ray#6358]
Hi, I want to use Ray to parallelize a GPU job, by using 3 GPU actors, as follows.

However, the result does not look good. It is even much slower than a serial single-GPU execution.

```
import psutil
import ray
import time
import numpy
import random

import face_alignment
from skimage import io


num_cpus = psutil.cpu_count(logical=False)
ray.init(num_cpus=num_cpus, num_gpus=3)

@ray.remote(num_gpus=1)
class GPUActor(object):
    def __init__(self):
        self.fa = face_alignment.FaceAlignment(
                face_alignment.LandmarksType._2D,
                flip_input=False, device='cuda')

        str = "This function is allowed to use GPUs {}.".format(ray.get_gpu_ids())
        print(str)
        self.preds = 0

    def proc(self, img):
        self.preds += self.fa.get_landmarks(img)[0].mean()

    def get_preds(self):
        return self.preds


gpuactors = [GPUActor.remote() for _ in range(3)]

begin = time.time()
for _ in range(20 * 3):
    image = io.imread('aflw-test.jpg')
    img_id = ray.put(numpy.array(image))
    gpuactors[_ % 3].proc.remote(img_id)

res = ray.get([actor.get_preds.remote() for actor in gpuactors ])
print(res)
end = time.time()
print('time', end- begin)

```

**Comment by robertnishihara** [comment]
This doesn't run for me because of the error `ModuleNotFoundError: No module named 'face_alignment'`.

I'd suggest looking at the Ray timeline. https://ray.readthedocs.io/en/latest/troubleshooting.html#visualizing-tasks-in-the-ray-timeline

Also, how long are the tasks that you are trying to parallelize? If they each take just a couple milliseconds, then you may not see a speedup.

**Comment by kwanUm** [comment]
This happens to me as well. Very weird.. After measuring each step, it seems like the model inference for face_alignment just runs slower in the context of ray.

I'm not sure why this happens, perhaps the library is using behind the scenes something that ray doesn't assumes?

@robertnishihara  you can install the face_alignment lib by "pip install face_alignment"

**Comment by Arjunsankarlal** [comment]
I am also facing the same issue here. What I am trying is,

```
@ray.remote
class Test:
    def __init__(self):
        try:
            self.model = loading large Neural Network model here

    @ray.method(num_return_vals=1)
    def predict(self, inputs):
        st = perf_counter()
        answer = self.model(*inputs)
        print(f'Time taken is {perf_counter()-st}')
        return answer
```

This is slower, in my case it takes 0.85sec, which when normally run as,

```
model = NNModel()
st = perf_counter()
answer = model(*inputs)
print(f'Time taken is {perf_counter()-st}')
```
This takes 0.4sec only. After a lot of debugging I found that, while serially executing as above, uses 3 cores for computing (Found it in the activity monitor in mac) and while using with Actor only one core is being utilized. So I tried setting multiple CPU cores for Actor by annotating with ```@remote(num_cpus=3)``` but this makes no change and I still see only one core being utilized for inferencing. 

My doubt here is since I am using a stateful actor (By loading the model at initialization) does it takes only one CPU for processing, so is this expected or I am I doing something wrong here ? 

Any help would be much appreciated!
